ext/rugged/rugged_remote.c

```diff
@@ -181,6 +181,17 @@ void rugged_remote_init_callbacks_and_payload_from_options(
 	}
 }
 
+static int parse_prune_type(VALUE rb_prune_type)
+{
+	if (rb_prune_type == Qtrue) {
+		return GIT_FETCH_PRUNE;
+	} else if (rb_prune_type == Qfalse) {
+		return GIT_FETCH_NO_PRUNE;
+	} else {
+		return GIT_FETCH_PRUNE_UNSPECIFIED;
```

Comment:
Yes, Rugged tries to be strict with type checking. A value of `"foobar"` should not result into a valid default because it's bogus.